apps/ledger-live-mobile/src/components/DeviceAction/rendering.tsx

```diff
@@ -894,17 +900,44 @@ export function renderLoading({
 }
 
 export function renderExchange({
+  swapRequest,
   exchangeType,
   t,
   device,
   theme,
 }: RawProps & {
+  swapRequest: {
+    provider: string;
+    selectedDevice: Device;
+    transaction: Transaction;
+    exchangeRate: ExchangeRate;
+    exchange: ExchangeSwap;
+    colors: T["colors"];
+    theme: typeof theme;
```

Comment:
Why not use the types `SettingsState`, `WalletState` and `Theme` ?

Reply:
done added, except theme as i need only 
```
theme: "light" | "dark" | undefined
```

Reply:
In that case you can do this instead, that way the type will be explicit and more readable
```suggestion
    theme?: "light" | "dark";
```

Reply:
Small detail: sometimes you cannot use the `?` with our current strict mode as optional params cannot receive an undefined value